consensus/propagation/reactor_test.go

```diff
@@ -87,105 +87,78 @@ func TestHandleHavesAndWantsAndRecoveryParts(t *testing.T) {
 			{Hash: cmtrand.Bytes(32)},
 			{Hash: cmtrand.Bytes(32)},
 		},
 	}
 
+	height, round := int64(10), int32(1)
+
 	// adding the proposal manually so the haves/wants and recovery
 	// parts are not rejected.
 	p := types.Proposal{
 		BlockID: types.BlockID{
-			Hash:          nil,
-			PartSetHeader: types.PartSetHeader{Total: 30},
+			Hash:          cmtrand.Bytes(32),
+			PartSetHeader: psh,
 		},
-		Height: 10,
-		Round:  1,
+		Height: height,
+		Round:  round,
 	}
 	baseCompactBlock.Proposal = p
+
 	added, _, _ := reactor1.AddProposal(baseCompactBlock)
 	require.True(t, added)
-
-	p2 := types.Proposal{
-		BlockID: types.BlockID{
-			Hash:          nil,
-			PartSetHeader: types.PartSetHeader{Total: 30},
-		},
-		Height: 10,
-		Round:  1,
-	}
-	baseCompactBlock.Proposal = p2
 	added, _, _ = reactor2.AddProposal(baseCompactBlock)
 	require.True(t, added)
-
-	p3 := types.Proposal{
-		BlockID: types.BlockID{
-			Hash:          nil,
-			PartSetHeader: types.PartSetHeader{Total: 30},
-		},
-		Height: 10,
-		Round:  1,
-	}
-	baseCompactBlock.Proposal = p3
-
 	added, _, _ = reactor3.AddProposal(baseCompactBlock)
 	require.True(t, added)
+
 	proof := merkle.Proof{LeafHash: cmtrand.Bytes(32)}
+	bm := bits.NewBitArray(10)
+	bm.Fill()
 
 	// reactor 1 will receive haves from reactor 2
 	reactor1.handleHaves(
 		reactor2.self,
 		&proptypes.HaveParts{
-			Height: 10,
-			Round:  1,
+			Height: height,
+			Round:  round,
 			Parts: []proptypes.PartMetaData{
-				{Index: 2, Proof: proof},
-				{Index: 3, Proof: proof},
-				{Index: 4, Proof: proof},
+				{Index: 0, Proof: proof},
 			},
 		},
-		true,
+		false,
 	)
 
-	haves, has := reactor1.getPeer(reactor2.self).GetHaves(10, 1)
+	haves, has := reactor1.getPeer(reactor2.self).GetHaves(height, round)
 	assert.True(t, has)
-	assert.Equal(t, int64(10), haves.Height)
-	assert.Equal(t, int32(1), haves.Round)
-	assert.Contains(t, haves.Parts, proptypes.PartMetaData{Index: 2, Proof: proof})
-	assert.Contains(t, haves.Parts, proptypes.PartMetaData{Index: 3, Proof: proof})
-	assert.Contains(t, haves.Parts, proptypes.PartMetaData{Index: 4, Proof: proof})
+	require.True(t, haves.GetIndex(0))
 
 	time.Sleep(400 * time.Millisecond)
 
-	// reactor 1 will gossip the haves with reactor 3
-	// check if the third reactor received the haves
 	r3State := reactor3.getPeer(reactor1.self)
 	require.NotNil(t, r3State)
 
-	r3Haves, r3Has := r3State.GetHaves(10, 1)
+	r3Haves, r3Has := r3State.GetHaves(height, round)
 	assert.True(t, r3Has)
-	assert.Contains(t, r3Haves.Parts, proptypes.PartMetaData{Index: 3, Proof: proof})
-	assert.Contains(t, r3Haves.Parts, proptypes.PartMetaData{Index: 4, Proof: proof})
-
-	// since reactor 3 received the haves from reactor 1,
-	// it will send back a want.
-	// check if reactor 1 received the wants
-	r1Want, r1Has := reactor1.getPeer(reactor3.self).GetWants(10, 1)
-	assert.True(t, r1Has)
-	assert.Equal(t, int64(10), r1Want.Height)
-	assert.Equal(t, int32(1), r1Want.Round)
```

Comment:
all the nodes are connected I think. therefore its not guaranteed they get the want from that peer

Reply:
yes, but we're setting the redundancy to 3. this means that they should all receive the haves from the proposer and send wants, no?

Reply:
nvm, checked and it's set to 2

Reply:
what was the logic with setting it to 2? it should be set to 1 since we have parity data

Reply:
it was set to 2 before we supported parity data